File: prob-3-largest-prime-factor/prob-3-largest-prime-factor.go
package main

import "fmt"

func largestPrimeFactor(n int64) int64 {
	var lpf int64 = 2
	for n > lpf {
		if n%lpf == 0 {
			n = n / lpf
			lpf = 2
		} else {
			lpf += 1
		}
	}
	return lpf
}

func main() {
	fmt.Println(largestPrimeFactor(600851475143))
}

File: prob-3-largest-prime-factor/prob-3-largest-prime-factor.py
def largestPrimeFactor(n):
    lpf = 2
    while n > lpf:
        if n % lpf == 0:
            n = n // lpf
            lpf = 2
        else:
            lpf += 1
    return lpf

print(largestPrimeFactor(600851475143))
